shap/plots/_beeswarm.py

```diff
@@ -542,11 +542,13 @@ def summary_legacy(
         #     out_names = shap_exp.output_names
 
     multi_class = False
-    if isinstance(shap_values, list):
+    shap_values_as_list = []
+    if isinstance(shap_values, np.ndarray) and shap_values.ndim == 3:
```

Comment:
Unfortunately, this won't work in general because `shap_values`  having 3 dimensions can also be from SHAP feature interactions. So at the very least, we'll have to check whether the second and third dimensions are the same. (But even then, it's not entirely foolproof.)

I think the solution here is not to patch summary_plot (it's a legacy function) any further, 1) it's trying to do quite a lot of things which makes it hard to maintain and reason about, 2) we run into conflicting conditions like these.  
Instead, I want to migrate over to the dedicated bar plot function(s) -- currently the `shap.plots.bar` doesn't support multiclass, I'm working on fixing that at the moment.